docs/guides/formulasC2/javascript.md

```diff
@@ -0,0 +1,61 @@
+---
+heading: Building Formulas
+seo: Javascript in Formulas | Formulas | Cloud Elements API Docs
+title: Javascript in Formulas
+description: How to Use Javascript in Formulas
+layout: sidebarleft
+apis: API Docs
+restContentVersion: referenceapi
+platform: formulas
+breadcrumbs: /docs/guides/home.html
+parent: Back to Guides
+order: 20
+---
+
+# Javascript in Formulas
+There are many step types that allow you to write your own custom Javascript.  The function signature for all JS-related step types looks like:
+
+```javascript
+/**
+ * @param  trigger The trigger that started this execution
+ * @param  steps   The list of steps that have been executed up until this point for this execution and all of their step execution values
+ * @param  info    Metadata about this formula
+ * @param  config  The configuration values set on the formula instance (config variables)
+ * @param  done    The callback function that you will need to call at the end of your script step
+ */
+function (trigger, steps, info, config, done) {
+  // your Javascript will be executed here
+}
+```
+
+Note the following when writing javascript in formulas:
+
+* For all scripts, Javascript `strict` mode is enforced.
+* You can use `console.log` to log things in your formula and help debug.
+* You can use `notify.email` to send an email notification.
+* ES6 is supported.
+* The function parameters are immutable, meaning they cannot be assigned to directly. In order to change an object or value passed into the function, first copy it to your own local variable and then make the necessary changes.
+
+## Functions
+
+* `console.log(str)`: Log something from the script. This logged value will be returned in an array called `console`, which will be available to see as a step execution value. Takes a `string` as a parameter.
+* `throw(str)`: Force a script to exit with an error message. The error message will be available to see as a step execution value. Takes a `string` as a parameter.
+* `notify.email(to, subject, body)`: Send an email notification directly from a Javascript step.  `to` can be a single email or a comma separated list of emails, `subject` is the subject of the email and `body` is the body of the email. A value will be returned in an array called `notify`, which will be available to see as a step execution value. Takes three `string` parameters. You can reference anything from the context when passing in `to`, `subject` or `body` in the same way you can access these variables elsewhere in the Javascript. For example:
+```
+notify.email(steps.previous-step.email, steps.previous-step.subject, steps.previous-step.emailPrefix + '<br>This is the main body.');
+```
+
+## Libraries
+
+* CE: Our custom library that provides some common functionality. It is not necessary to `require` this library, it is available by default.
```

Comment:
Can this be an `include` or shared across all of the different parts of our platform that allow JS? (elements, transformations, formulas, etc.)